apps/remote_control/test/lexical/remote_control/code_mod/extract_function_test.exs

```diff
@@ -0,0 +1,280 @@
+defmodule Lexical.RemoteControl.CodeMod.ExtractFunctionTest do
+    alias Lexical.RemoteControl.CodeMod.ExtractFunction
+
+    use ExUnit.Case
+    alias Sourceror.Zipper, as: Z
+
+    setup ctx do
+      if Map.has_key?(ctx, :no_setup) do
+        {:ok, []}
+      else
+        no = Map.get(ctx, :no, "")
+
+        {:ok,
+         quoted:
+           """
+           defmodule Baz#{no} do
+             def foo(one, two) do
+               three = 3
+               IO.inspect(one)
+               IO.inspect(two)
+               IO.inspect(three)
+               four = 4
+               IO.inspect(three)
+
+               IO.inspect(four: four,
+                 force_format_on_new_line_with_really_long_atom: true
+               )
+
+               # comment
+             end
+           end
+           """
+           |> Sourceror.parse_string!()}
+      end
+    end
+
+    describe "extract_function" do
+      @tag no: 1
+      test "extract one line to function", %{quoted: quoted} do
+        zipper = ExtractFunction.extract_function(Z.zip(quoted), 3, 3, "bar")
```

Comment:
This likely is a code action, rather than a codemod, and we have a lot of support for code actions marking ranges and marking code blocks, so you could build, for example

```elixir
use Lexical.CodeMod.Case, enable_ast_support: false
import Lexical.Test.CodeSigil
import Lexical.Test.RangeSupport

    start_supervised!({Document.Store, derive: [analysis: &Lexical.Ast.analyze/1]})
    :ok
  end

  def apply_code_mod(text, _ast, options) do
    range = options[:range]
    uri = "file:///file.ex"
    :ok = Document.Store.open(uri, text, 1)
    {:ok, document} = Document.Store.fetch(uri)

    [action] = ExtractFunction.actions(document, range, [])
    {:ok, action.changes.edits}
  end

  def extract_function(original_text) do
    {position, stripped_text} = pop_range(original_text)
    range = Range.new(position, position)
    modify(stripped_text, range: range)
  end

test "extracting..." do 
   {:ok, new_code} = 
    ~q[
       defmodule Bar do
          def foo(one, two) do
             «IO.inspect(one)
             IO.inspect(two)»
       end
     ] |> extract_function("bar")
    # the new code will run your codemod / code action and return the modified string, which you can 
    # assert against
 end
```

Note the french quotes («») around the range you'd like to extract.